components/locale/examples/filter_langids.rs

```diff
@@ -0,0 +1,54 @@
+// A sample application which takes a comma separated list of language identifiers,
+// filters out identifiers with language subtags different than `en` and serializes
+// the list back into a comma separated list in canonical syntax.
+use std::env;
+
+use icu_locale::{subtags, LanguageIdentifier};
+
+const DEFAULT_INPUT: &str =
+    "de, en-us, zh-hant, sr-cyrl, fr-ca, es-cl, pl, en-latn-us, ca-valencia, und-arab";
+
+fn filter_input(input: &str) -> String {
+    // 1. Parse the input string into a list of language identifiers.
+    let langids: Vec<LanguageIdentifier> = input
+        .split(',')
+        .filter_map(|s| s.trim().parse().ok())
+        .collect();
+
+    // 2. Filter for LanguageIdentifiers with Language subtag `en`.
+    let en_lang: subtags::Language = "en".parse().expect("Failed to parse language subtag.");
+
+    let en_langids = langids
+        .into_iter()
+        .filter(|langid| langid.language == en_lang);
+
+    // 3. Serialize the output.
+    let en_strs: Vec<String> = en_langids.map(|langid| langid.to_string()).collect();
+
+    en_strs.join(", ")
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let (_input, _output) = if let Some(input) = args.get(1) {
```

Comment:
Optional/minor Rust question: Would it have been possible for this to be written as:

```rust
    let _input = if let Some(input) = args.get(1) {
        input.as_str()
    } else {
        DEFAULT_INPUT
    };
    let _ouptut = filter_input(&input);
...
    println!("...", &_input, &_output);
```

I forgot if `println!` complains on `&String` or not.

Reply:
I like it! Applied! thank you!